content/docs/rfcs/32/README.md

```diff
@@ -0,0 +1,560 @@
+Another use case is to protect an anonymous messaging system against DDoS and spam attacks by containing messaging rate of users. 
+This latter use case is explained in [17/Waku-RLN-Relay RFC](/spec/17).
+
+
+# Flow
+
+The users participate in the protocol by first registering to an application-defined group referred by the _membership group_.
+Registration to the group is mandatory for signaling in the application. 
+After registration, group members can generate Zero-knowledge Proof of membership for their signals and can participate in the application. 
+Usually, the membership requires a financial or social stake which 
+is beneficial for the prevention of Sybil attacks and double-signaling.
+Group members are allowed to send one signal per epoch.
+If a user generates more signals than allowed in an epoch, 
+the user risks being slashed - by revealing his membership secret credentials. secret and revealing his identity commitment. 
+If the financial stake is put in place, the user also risks his stake being taken.
+
+Generally the flow can be described by the following steps:
+
+1. Registration
+2. Signaling
+3. Verification and slashing
+
+
+## Registration
+
+Depending on the application requirements, the registration can be implemented in different ways, for example: 
+- centralized registrations, by using a central server
+- decentralized registrations, by using a smart contract
+
+What is important is that the user's identity commitment is stored in a merkle tree, 
+and the users can obtain a merkle proof proving that they are part of the group.
+
+Also depending on the application requirements, 
+usually a financial or social stake is introduced.
+
+An example for financial stake is: For each registration a certain amount of ETH is required.
+An example for social stake is using InterRep as a registry - 
+users need to prove that they have a highly reputable social media account.
+
+### Implementation notes
+
+#### User identity
+
+The user's identity is composed of:
+
+```
+{
+    identity_secret: [identity_nullifier, identity_trapdoor],
+    identity_secret_hash: poseidonHash(identity_secret),
+    identity_commitment: poseidonHash([identity_secret_hash])
+}
+```
+
+For registration, the user needs to submit their `identity_commitment` (along with any additional registration requirements) to the registry. 
+Upon registration, they should receive `leaf_index` value which represents their position in the merkle tree. 
+Receiving a `leaf_index` is not a hard requirement and is application specific. 
+The other way around is the users calculating the `leaf_index` themselves upon successful registration.
+
+## Signaling
+
+After registration, 
+the users can participate in the application by sending signals to the other participants in a decentralised manner or to a centralised server. 
+Along with their signal, 
+they need to generate a ZK-Proof by using the circuit with the specification described above. 
```

Comment:
there is no circuit specification `above`, you may put a link to the corresponding section.